crates/sui/src/client_commands.rs

```diff
@@ -1173,18 +1266,57 @@ impl SuiClientCommands {
                 type_args,
                 gas,
                 gas_budget,
+                dry_run,
                 gas_price,
                 args,
                 serialize_unsigned_transaction,
                 serialize_signed_transaction,
             } => {
-                let tx_data = construct_move_call_transaction(
-                    package, &module, &function, type_args, gas, gas_budget, gas_price, args,
-                    context,
+                let tx_kind = construct_move_call_transaction_kind(
+                    package, &module, &function, type_args, args, context,
                 )
                 .await?;
+                let gas_owner = context.try_get_object_owner(&gas).await?;
+                let signer = gas_owner.unwrap_or(context.active_address()?);
+                let gas_price = gas_price.unwrap_or(context.get_reference_gas_price().await?);
```

Comment:
To avoid the API call if we don't need it:

```suggestion
                let gas_price = if let Some(gas_price) {
                    gas_price
                } else 
                    context.get_reference_gas_price().await?
                };
```